Task2 selenium we

BorysFe opened the pull request.

Commit:
Added:
- Opening url
- Authorization by user.
- Test - Asserting title. Need to add some waiter for page before assert.

Commit:
3 new test for registration

Commit:
Create new tests:
- 4 test for boards
- 3 test for authorization
Moved driver initialization in methods.

Commit:
Changed the project structure
Moved the driver to the before class for tests.
Used different ways to define elements

edward-astapov commented on src/main/java/com/trello/BoardPage.java:
```diff
@@ -0,0 +1,102 @@
+    }
+
+    public void newBoardTitle() {
+
```
?

edward-astapov commented on src/main/java/com/trello/BoardPage.java:
```diff
@@ -0,0 +1,102 @@
+    }
+
+    public String getClosedBoardMessage() {
+    return closeBoardMessage.getText();
```
code format

BorysFe replied:
Done

edward-astapov commented on src/main/java/com/trello/LogInPage.java:
```diff
@@ -0,0 +1,73 @@
+    private WebElement erdddrorMessage;
+
+
+    String emailUser1 = "fesenko.b@icloud.com";
```
Remove credentials from page. All values should be passed from test.

BorysFe replied:
ok

BorysFe replied:
Done

edward-astapov commented on src/main/java/com/trello/LogInPage.java:
```diff
@@ -0,0 +1,73 @@
+        openLogInPageButton.click();
+    }
+
+    public void logInExistedUser1() {
```
Need single login method that receives email and password from test

BorysFe replied:
Done

edward-astapov commented on src/test/java/com/trello/authorization/AuthorisationTest.java:
```diff
@@ -0,0 +1,53 @@
+public class AuthorisationTest {
+
+    private ChromeDriver driver;
+    LogInPage logInPage = new LogInPage(driver);
```
Do not create page object as test field. Use separate pages in each test

BorysFe replied:
Done, if I understand right.

edward-astapov commented on src/test/java/com/trello/boards/NewBoardTest.java:
```diff
@@ -0,0 +1,106 @@
+
+    @AfterMethod
+    public void signOffFromTrello() {
+//        System.setProperty("webdriver.chrome.driver", "C://Users//Office//Downloads//chromedriver_win32 (1)//chromedriver87.exe");
```
If code is not needed - please remove it

BorysFe replied:
Removed

edward-astapov commented on src/test/java/com/trello/boards/NewBoardTest.java:
```diff
@@ -0,0 +1,106 @@
+        Actions action = new Actions(driver);
+
+        WebElement userBoard = driver.findElement(By.xpath(".//a[@title= 'Borys_Tech-Stack']"));
+        String boardTitle = driver.findElement(By.xpath(".//input[@class= 'board-name-input js-board-name-input']"))
```
If page object used - it is not recommended to do findElement in test class. Move this to page object, if possible

BorysFe replied:
Done

edward-astapov commented on src/test/java/com/trello/boards/NewBoardTest.java:
```diff
@@ -0,0 +1,106 @@
+        boardPage.openMemberMenu();
+        boardPage.addNewBoard(newBoardTitle);
+
+        Assert.assertEquals(boardPage.boardTitle.getAttribute("value"), newBoardTitle, "The board title is wrong");
```
Move methods like this to page classes

BorysFe replied:
Done

edward-astapov commented on src/test/java/com/trello/registration/RegistrationTest.java:
```diff
@@ -0,0 +1,80 @@
+public class RegistrationTest {
+
+    private ChromeDriver driver;
+    RegistrationPage registrationPage = new RegistrationPage(driver);
```
Use separate pages in each test

edward-astapov commented on src/test/java/com/trello/registration/RegistrationTest.java:
```diff
@@ -0,0 +1,80 @@
+    @BeforeClass
+    public void webDriver() {
+        System.setProperty("webdriver.chrome.driver", "C://Users//Office//Downloads//chromedriver_win32 (1)//chromedriver87.exe");
+//        driver = new ChromeDriver();
```
?

edward-astapov commented on src/test/java/com/trello/registration/RegistrationTest.java:
```diff
@@ -0,0 +1,80 @@
+
+    @Test
+    public void privacyPolicyTextCheck() {
+        System.setProperty("webdriver.chrome.driver", "C://Users//Office//Downloads//chromedriver_win32 (1)//chromedriver87.exe");
```
We decide to move this in @BeforeClass

edward-astapov commented on src/main/java/com/trello/BoardPage.java:
```diff
@@ -0,0 +1,102 @@
+    private WebElement newBoardSubmit;
+
+    @FindBy (xpath = ".//input[@class= 'board-name-input js-board-name-input']")
+    public WebElement boardTitle;
```
ALL WebElements in Page classes should be **private**
After changing all elements to private - please update affected code, by moving some methods from test to page class (see comments in tests for more details)

BorysFe replied:
ok

edward-astapov reviewed (changes requested):
Please fix comments

Commit:
More changed the project structure
All pageObjects locates in the page classes.

edward-astapov commented on src/main/java/com/trello/BoardPage.java:
```diff
@@ -0,0 +1,126 @@
+        return deleteBoardMessage.getText();
+    }
+
+    public String getCustomUserBoardTitle(String eee) {
```
'eee' is not valid  variable name

edward-astapov commented on src/main/java/com/trello/BoardPage.java:
```diff
@@ -0,0 +1,126 @@
+    @FindBy (xpath = ".//div[@class= 'js-react-root']//h1")
+    private WebElement deleteBoardMessage;
+
+    private final String userBoard = (".//a[@title= '%s']");
```
You can remove brackets

Commit:
Change format code and add missing PageObject.

edward-astapov commented on src/main/java/com/trello/LogInPage.java:
```diff
@@ -0,0 +1,50 @@
+
+    public LogInPage(WebDriver driver) {
+        PageFactory.initElements(driver, this);
+        driver.get("https://trello.com");
```
Please avoid this in future.

edward-astapov commented on src/main/java/com/trello/RegistrationPage.java:
```diff
@@ -0,0 +1,88 @@
+
+    public RegistrationPage(WebDriver driver) {
+        PageFactory.initElements(driver, this);
+        driver.get("https://trello.com");
```
Same here. Avoid extra code in constructors

edward-astapov commented on src/test/java/com/trello/authorization/AuthorisationTest.java:
```diff
@@ -0,0 +1,67 @@
+    public void signOffFromTrello() {
+        BoardPage boardPage = new BoardPage(driver);
+        boardPage.logOutIfAuthorised();
+        ;
```
?

edward-astapov commented on src/test/java/com/trello/authorization/AuthorisationTest.java:
```diff
@@ -0,0 +1,67 @@
+    @BeforeClass
+    public void webDriver() {
+        System.setProperty("webdriver.chrome.driver", "C://Users//Office//Downloads//chromedriver_win32 (1)" +
+                "//chromedriver87.exe");
```
Avoid line breaks like this

edward-astapov reviewed (changes requested):
Need works

Commit:
Changed the position of the environment opening
Editing by comments.
Added additional check for user logOut - if there is a logout button, but it is not on the screen.
